site/src/pages/components/openable.explainer.mdx

```diff
@@ -0,0 +1,295 @@
+This allows the element to maintain find-in-page support if it's opened and collapsed again.
+
+```html
+<div id="findable-openable" hidden="until-found" openable>
+  Searchable hidden contents
+</div>
+```
+
+### Rendering
+
+The UA stylesheet for openables will look like this:
+
+```css
+/** Hide the contents of the openable when it's not open */
+[openable]:not([hidden=until-found]):not(:open) {
+  display: none;
+}
+
+/** Revert hidden=until-found's style when the openable is open */
+[openable][hidden=until-found]:open {
+  content-visibility: revert;
+}
+```
+
+## IDL Attribute and Feature Detection
+
+The `openable` content attribute will be [reflected](https://html.spec.whatwg.org/#reflect) as a boolean IDL attribute:
+
+```
+[Exposed=Window]
+partial interface HTMLElement {
+  attribute boolean openable;
+
+  undefined openOpenable();
+  undefined closeOpenable();
+  boolean toggleOpenable(optional ToggleOpenableOptions options = {});
+}
+
+dictionary ToggleOpenableOptions {
+  boolean force;
+}
+```
+
+This not only allows developer ease-of-use from JavaScript, but also allows for a feature detection mechanism:
+
+```javascript
+function supportsOpenable() {
+  return HTMLElement.prototype.hasOwnProperty('openable')
+}
+```
+
+## Events
+
+An event is fired synchronously when an openable is open or close. This event can be used, for example, to populate content for the openable just in time before it is opened, or update server data when it closes. The event provides a `currentState` and `newState` for the openable. The value of these properties can either be "open" or "closed". The events looks like this:
+
+```javascript
+openable.addEventListener('beforetoggle', (beforeToggleEvent) => {
+  if (beforeToggleEvent.currentState === 'closed') console.info('Openable is closed')
+  if (beforeToggleEvent.newState === 'open') console.info('Openable is being shown')
+})
+```
+
+The `beforetoggle` event is cancellable when the `newState` is equal to "open". Doing so keeps the openable from being opened. You can't cancel the closing of an openable. The `beforetoggle` event is fired synchronously.
+
+Additionally an asynchronous non-cancellable `toggle` event is fired.
+
+## Focus Management
+
+TODO
+
+## Accessibility / Semantics
+
+Since the `openable` content attribute can be applied to any element, and this only impacts the element's presentation, this addition does not have any direct semantic or accessibility impact. The element with the `openable` attribute will keep its existing semantics and AOM representation. For example, `<article openable>...</article>` will continue to be exposed as an implicit `role=article`. Similarly, ARIA can be used to modify accessibility mappings in [the normal way](https://w3c.github.io/html-aria/), for example `<div openable role=note>...</div>`.
```

Comment:
```suggestion
The `openable` content attribute can be applied to any element, where in the hidden or "collapsed" state, the element will be hidden from the accessibility tree.  Upon invoking. the element will become available both visually, and via the browser's accessibility tree.  Beyond this behavior, the element with the `openable` attribute will otherwise keep its existing semantics and AOM representation. For example, `<article openable>...</article>` will continue to be exposed as an implicit `role=article`. Similarly, ARIA can be used to modify accessibility mappings [per the allowances defined in the ARIA in HTML specification](https://w3c.github.io/html-aria/).  For example `<div openable role=note>...</div>`.
```